### DISCUSSION TITLE
install on WI-DG?

### DISCUSSION THREAD

**Discussion by scottshanafelt** [mrene/minidsp-rs#414] [Q&A]
Maybe this is addressed elsewhere but a quick search didn't find anything.  Can minidsp-rs be installed on the WI-DG itself?  I was on the command line of the WI-DG itself (specifically to reconfigure the ethernet settings) and being that it's basically linux/busybox I was wondering why we couldn't install minidsp-rs directly?  Has anyone tried it?

**Comment by mrene** [comment]
I had no idea the wi-dg ran linux, but I don't see any reason why it wouldn't work.

One thing I'd be worried about is two processes interacting with the device over USB HID, but you could configure minidsp-rs to use a [static device](https://minidsp-rs.pages.dev/daemon/?highlight=WI-DG#example) pointing to localhost and expose an HTTP API on the wi-dg itself. 

What architecture/distro does it use (`uname -a`, `cat /etc/os-release`)? How much free disk space is on it (`df -h`)?

**Comment by scottshanafelt** [comment]
The login banner says it is OpenWrt


***@***.***:~# uname -a
Linux miniDSP-Wi-DG 3.18.91 #7 Wed May 30 15:26:34 HKT 2018 mips GNU/Linux


***@***.***:~# df -h
Filesystem                Size      Used Available Use% Mounted on
rootfs                   10.6M    804.0K      9.8M   7% /
/dev/root                20.0M     20.0M         0 100% /rom
tmpfs                    61.7M    236.0K     61.5M   0% /tmp
/dev/mtdblock6           10.6M    804.0K      9.8M   7% /overlay
overlayfs:/overlay       10.6M    804.0K      9.8M   7% /
tmpfs                   512.0K         0    512.0K   0% /dev


***@***.***:~# grep MemTotal /proc/meminfo
MemTotal:         126444 kB


Yes agree on using IP loopback to talk to device rather than USB.  I like this whole idea over running the daemon as a docker elsewhere.  My goal here is to integrate this using your daemon to my crestron system.  Anyway if I do get around to trying this I'll update here (this is for my personal system so in no rush).

Scott

**Reply by mrene** [reply]
You'd need to get it compiled for mips (the [cross](https://github.com/cross-rs/cross) project lists support for `mips-unknown-linux-gnu` and `mips-unknown-linux-musl`, one of these should work - and `musl` would be my guess because of busybox). These images won't support libusb by default, unless you change the [docker image](https://github.com/mrene/minidsp-rs/blob/dev/scripts/Dockerfile.aarch64) - but in your case since you want to use tcp only, you could  get `minidspd` to build without the usb dependency, with some minor changes.

The 10mb writable space might be something to keep an eye on. It should fit... the stripped binary is around 5mb as it is now. 

Let me know how it goes!

**Reply by penguinpages** [reply]
Coming back to poke this thread.   I have a miniDSP Flex, but the only way to control it is via laptop directly plugged into it.   The remote app for phone is a build out of "remote" so just basic keys, but what I am struggling with is per posting I made, that never was responded to .. but seems forum timed it out or took it down.  Issue is that I need ONE button to set mode. Not just change "preset button" but also to change Input from Bluetooth to Analog etc..    The MiniDSP can't even connect withe app anymore, I think due to old RSA token that the phones /apps reject.    

 Question
1) Can I use microsd slot, load up some form of boot OS, and boot more functional bridge to control my Flex?
2) Does anyone know where logs are on this unit such that I can root cause application error

![image](https://github.com/mrene/minidsp-rs/assets/17153876/c26b5f03-03d1-4aea-9bbc-afe1f90bf789)

```
root@miniDSP-Wi-DG:/etc/dropbear# cd /var/
root@miniDSP-Wi-DG:/tmp# ls -alh
drwxrwxrwt   19 root     root         500 Feb 23 00:46 .
drwxr-xr-x    1 root     root           0 Aug 26  2022 ..
drwxr-xr-x    2 root     root          40 Aug 26  2022 .jail
drwx------    2 root     root          40 Aug 26  2022 .uci
-rw-r--r--    1 root     root           4 Aug 26  2022 TZ
-rw-r--r--    1 root     root           4 Aug 26  2022 apcli-apcli0.pid
-rw-r--r--    1 root     root           0 Aug 26  2022 dhcp.leases
drwxr-xr-x    2 root     root          40 Aug 26  2022 dnsmasq.d
drwxr-xr-x    2 root     root          80 Aug 26  2022 etc
drwxr-xr-x    2 root     root          60 Aug 26  2022 hosts
drwxr-xr-x    3 root     root          60 Aug 26  2022 lib
drwxrwxrwx    2 root     root         360 Feb 23 00:55 lock
drwxr-xr-x    2 root     root         120 Aug 26  2022 log
dr-xr-xr-x    2 root     root          40 Aug 26  2022 mounts
-rw-r--r--    1 root     root        3.7K Aug 26  2022 mt7628.dat
drwxr-xr-x    2 root     root          60 Aug 26  2022 nmbd
drwxr-xr-x    2 root     root          40 Feb 23 00:37 opkg-lists
drwxr-xr-x    2 root     root          40 Jan  1  1970 overlay
-rw-r--r--    1 root     root          86 Feb 23 00:46 resolv.conf
-rw-r--r--    1 root     root          86 Aug 26  2022 resolv.conf.auto
drwxrwxrwx    6 root     root         300 Feb 23 00:39 run
drwxrwxrwt    2 root     root          40 Jan  1  1970 shm
drwxrwxrwx    2 root     root          60 Aug 26  2022 state
drwxr-xr-x    2 root     root          80 Jan  1  1970 sysinfo
drwxr-xr-x    3 root     root          60 Feb 23 00:37 usr
root@miniDSP-Wi-DG:/tmp# cd log/
root@miniDSP-Wi-DG:/tmp/log# ls
lastlog   log.nmbd  log.smbd  wtmp
root@miniDSP-Wi-DG:/tmp/log# ls -alh
drwxr-xr-x    2 root     root         120 Aug 26  2022 .
drwxrwxrwt   19 root     root         500 Feb 23 00:46 ..
-rw-r--r--    1 root     root           0 Aug 26  2022 lastlog
-rw-r--r--    1 root     root           0 Aug 26  2022 log.nmbd
-rw-r--r--    1 root     root           0 Aug 26  2022 log.smbd
-rw-r--r--    1 root     root           0 Aug 26  2022 wtmp
root@miniDSP-Wi-DG:/tmp/log#
```

Logs are empty so hard to root cause.   Phone just says "Connection closed" when pointing at IP.
